File: console/src/types/strategy.ts
export interface StrategyKpi {
  label: string
  value: string
  hint: string
  kind: string | null
}

export interface SettledBet {
  when: string
  market: string
  clv: string
  result: string
  kind: string
}

export interface StrategyResponse {
  strategy_id: string
  name: string
  version: string
  mode: string
  enabled: boolean
  kpis: StrategyKpi[]
  settled: SettledBet[]
}

File: api/models/strategy.py
from __future__ import annotations

from pydantic import BaseModel


class StrategyKpi(BaseModel):
    label: str
    value: str
    hint: str
    kind: str | None = None


class SettledBet(BaseModel):
    when: str
    market: str
    clv: str
    result: str
    kind: str


class StrategyResponse(BaseModel):
    strategy_id: str
    name: str
    version: str
    mode: str
    enabled: bool
    kpis: list[StrategyKpi]
    settled: list[SettledBet]
